### PR TITLE
Update helmchart

### PR DESCRIPTION
**Pull request by aeimer**
I updated the helmchart to be actually usable.

I used `helm create enroute` as a base.

### CODE REVIEW

**Comment by chintan8saaras** [comment]
Thank you for your contribution @aeimer !

We will review the changes and verify this at our end.

If you can update this PR with the following information, it'll help -

1. Any challenges you faced with old helm chart.
2. An output with --dry-run flag when invoking helm. This will help us compare with the old version
3. The tests you have run with the modified chart

Thanks!

**Comment by aeimer** [comment]
Hi @chintan8saaras 
1. The chart
    - does not support other namespaces than `enroutedemo` which is quite ugly
    - it creates the namespace in any case; helm-chart should not create their own namespaces; better use `helm install [...] --create-namespace` or create the NS externally (like we do)
    - No HPA in place
    - Not all options that may be interesting could be adapted or were not working
    - No proper labels were used
    - namespaces were set in the yamls
 2. You can create that on your own:
     ```
     curl -o enroute.tgz https://raw.githubusercontent.com/aeimer/enroute/update-helmchart/helm-chart/enroute-0.5.0.tgz
     helm install -n test someenroute enroute.tgz --dry-run
     ```
3. I had to write this to get your service up and running for testing/evaluating purposes. So I couldn't test it extensively.

**Comment by chintan8saaras** [comment]
Created #67 to track this issue

File: helm-chart/enroute/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -1,80 +1,12 @@
-{{- if .Values.enrouteService.rbac.create -}}
+{{- if .Values.serviceAccount.create -}}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: enroutedemo
-  namespace: {{ .Values.globalSettings.namespace.name }}
+  name: {{ include "enroute.serviceAccountName" . }}
```
Comment (chintan8saaras) [inline helm-chart/enroute/templates/serviceaccount.yaml:5]: Do we need the namespace here -

```
namespace: {{ .Release.Namespace }}
```

Note, that the ```ClusterRoleBinding``` references the ```ServiceAccount``` along with namespace on line 67 in roles.yaml

File: helm-chart/enroute/templates/roles.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+metadata:
+  name: {{ include "enroute.fullname" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ include "enroute.fullname" . }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ include "enroute.fullname" . }}
+    namespace: {{ .Release.Namespace }}
```
Comment (chintan8saaras) [inline helm-chart/enroute/templates/roles.yaml:67]: We use the namespace here to reference the ```ServiceAccount``` but the creation of ```ServiceAccount``` doesn't have the namespace in ```serviceaccount.yaml```

**Review by chintan8saaras — CHANGES_REQUESTED** [review]

**Comment by chintan8saaras** [comment]
@aeimer Can you make that one change and resubmit? I was able to test installation with the new chart after the changes. Also please add a DCO sign-off similar to this? - https://cert-manager.io/docs/contributing/sign-off/

Thanks!

**Comment by aeimer** [comment]
@chintan8saaras I added a new commit with sign off

File: helm-chart/enroute/templates/serviceaccount.yaml (hunk 1 above)
Comment (aeimer) [reply helm-chart/enroute/templates/serviceaccount.yaml:5]: The namespace is set implicitly by helm. Eg the command `helm template -n test someenroute helm-chart/enroute-0.5.0.tgz` contains the NS `test` and everything will be deployed to that namespace. If you have the need to deploy object to not the "default" NS defined in the helm command, then you can set an explicit namespace. But its not required by default.

The ClusterRoleBinding is a clusterwide resource and SA are namespaced. Thats why we need to set the namspace there explicitly.

File: helm-chart/enroute/templates/roles.yaml (hunk 2 above)
Comment (aeimer) [reply helm-chart/enroute/templates/roles.yaml:67]: See above

**Pull request merged**